activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -90,7 +90,6 @@ def quote_string(string)
       #++
 
       def active?
-        return false unless @connection
         @connection.ping
```

Comment:
Nope, the gem will return false in that case (https://github.com/brianmario/mysql2/blob/0.4.4/ext/mysql2/client.c#L1016-L1020)